server/index.js

```diff
@@ -6,79 +6,91 @@ const express = require('express');
 const app = express();
 const axios = require('axios');
 const utils = require('../database/utils.js');
-const db = require('../database/index.js');
+const db = require('../database/index.js'); // remove once MVC is set up
+const controller = require('../controller/index.js');
 
 const port = 3004;
 
 app.use('/items/:itemId', express.static('client/dist'));
 app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
 
-// Get all items from db
+// Get all items from db (not used on front end. built for CRUD demonstration)
 app.get('/shopping/items', (req, res) => {
-  db.model.getAll((err, data) => {
-    if (err) {
-      console.log('failed to get all data:', err);
-      res.status(404).send(err);
-    } else {
-      console.log('successfully retrieved all data');
-      res.status(200).send(data.rows);
-    }
+  controller.shopping.getAll()
+  .then((shoppingData) => {
+    console.log('Successfully retrieved all data')
+    res.status(200).send(shoppingData.rows);
+  })
+  .catch((err) => {
+    res.status(404).send(err);
   });
 });
 
 // Create new item
 app.post('/shopping/items', (req, res) => {
-  db.model.post(req.body, (err, data) => {
-    if (err) {
-      console.log(`failed to add new item: ${err}`);
-      res.status(404).send(err);
-    } else {
-      console.log('successfully added new item');
-      res.status(200).send(data);
-    }
-  });
-});
+  controller.shopping.create(req.body)
```

Comment:
do you want to be sending the whole body here? or should you check for the properties you need first? i think it works either way - might just change the error handling

Reply:
Req.body actually wound up being only the info I send from Postman (i.e. only the object w/ exact data I need). I might have to adjust if I wind up needing post functionality on the front end.